homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -0,0 +1,509 @@
+            vol.All(cv.time_period, vol.Clamp(min=MIN_UPDATE_INTERVAL))),
+        vol.Optional(CONF_CHARGING_INTERVAL,
+                     default=DEFAULT_CHARGING_INTERVAL): (
+                         vol.All(cv.time_period,
+                                 vol.Clamp(
+                                     min=MIN_UPDATE_INTERVAL))),
+        vol.Optional(CONF_CLIMATE_INTERVAL,
+                     default=DEFAULT_CLIMATE_INTERVAL): (
+                         vol.All(cv.time_period,
+                                 vol.Clamp(
+                                     min=MIN_UPDATE_INTERVAL))),
+        vol.Optional(CONF_FORCE_MILES, default=False): cv.boolean
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+LEAF_COMPONENTS = [
+    'sensor', 'switch', 'binary_sensor', 'device_tracker'
+]
+
+SIGNAL_UPDATE_LEAF = 'nissan_leaf_update'
+
+SERVICE_UPDATE_LEAF = 'update'
+ATTR_VIN = 'vin'
+
+UPDATE_LEAF_SCHEMA = vol.Schema({
+    vol.Required(ATTR_VIN): cv.string
+})
+
+
+async def async_setup(hass, config):
+    """Set-up the Nissan Leaf component."""
+    import pycarwings2
+
+    async def handle_update(service):
+        # It would be better if this was changed to use nickname, or
+        # an entity name rather than a vin.
+        vin = service.data.get(ATTR_VIN, '')
+
+        if vin in hass.data[DATA_LEAF]:
+            data_store = hass.data[DATA_LEAF][vin]
+            async_track_point_in_utc_time(hass, data_store.async_update_data,
+                                          utcnow())
+            return True
+
+        _LOGGER.debug("Vin %s not recognised for update.", vin)
+        return False
+
+    async def async_setup_leaf(car_config):
+        """Set-up a car."""
+        _LOGGER.debug("Logging into You+Nissan...")
+
+        username = car_config[CONF_USERNAME]
+        password = car_config[CONF_PASSWORD]
+        region = car_config[CONF_REGION]
+        leaf = None
+
+        async def leaf_login():
+            nonlocal leaf
+            sess = pycarwings2.Session(username, password, region)
+            leaf = sess.get_leaf()
+
+        try:
+            # this might need to be made async (somehow) causes
+            # homeassistant to be slow to start
+            await hass.async_add_job(leaf_login)
+        except(RuntimeError, urllib.error.HTTPError):
+            _LOGGER.error(
+                "Unable to connect to Nissan Connect with "
+                "username and password")
+            return False
+        except KeyError:
+            _LOGGER.error(
+                "Unable to fetch car details..."
+                " do you actually have a Leaf connected to your account?")
+            return False
+        except pycarwings2.CarwingsError:
+            _LOGGER.error(
+                "An unknown error occurred while connecting to Nissan: %s",
+                sys.exc_info()[0])
+            return False
+
+        _LOGGER.info("Successfully logged in and fetched Leaf info")
+        _LOGGER.warning(
```

Comment:
Don't double log... even if at different levels (info - warning)

Reply:
Thanks.  I removed the _LOGGER.info("Successfully....") logging in commit https://github.com/home-assistant/home-assistant/pull/19786/commits/c4ced4b14b56fe0fe65d3e5caedb3299dc15252d